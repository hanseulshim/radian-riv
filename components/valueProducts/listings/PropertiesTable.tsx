import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'
import HistoricalListingModal from './HistoricalListingModal'
import { formatPrice } from 'utils'
import { PropertyInterface } from 'api'
import GalleryModal from './GalleryModal'

interface Props {
  tableData: PropertyInterface[]
  type: string
}

export default function PropertiesTable({ tableData, type }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])
  const [selectedProperty, setSelectedProperty] = useState<PropertyInterface>(
    null
  )
  const [historicalListingModal, setHistoricalListingModal] = useState(false)
  const [galleryModal, setGalleryModal] = useState(false)
  const fetchData = useCallback(async () => {
    setColumns([
      {
        Header: 'Street Address',
        accessor: 'address',
        align: 'right',
        width: 210
      },
      {
        Header: 'City',
        accessor: 'city',
        align: 'right',
        width: 200
      },
      {
        Header: 'Zip',
        accessor: 'zip',
        align: 'right',
        width: 100
      },
      {
        Header: 'Bed',
        accessor: 'bed',
        align: 'right',
        width: 75
      },
      {
        Header: 'Bath',
        accessor: 'bath',
        align: 'right',
        width: 75
      },
      {
        Header: 'SqFt',
        accessor: 'sqft',
        align: 'right',
        width: 90
      },
      {
        Header: 'Garage',
        accessor: 'garage',
        align: 'right',
        width: 75
      },
      {
        Header: 'Lot Size',
        accessor: 'lotSize',
        align: 'right',
        width: 90
      },
      {
        Header: 'Year Built',
        accessor: 'yearBuilt',
        align: 'right',
        width: 110
      },
      {
        Header: 'Target Distance',
        accessor: 'targetDistance',
        align: 'right',
        width: 100
      },
      {
        Header: 'List Date',
        accessor: 'listingDate',
        align: 'right'
      },
      {
        Header: 'List Price',
        accessor: row => formatPrice(row.listingPrice),
        align: 'right',
        width: 175
      },
      {
        Header: 'COE Date',
        accessor: 'coeDate',
        align: 'right'
      },
      {
        Header: 'Sold Price',
        accessor: row => formatPrice(row.soldPrice),
        align: 'right',
        width: 175
      },
      {
        Header: 'ACT DOM',
        accessor: 'actDom',
        align: 'right'
      },
      {
        Header: 'TOT DOM',
        accessor: 'totDom',
        align: 'right'
      },
      {
        Header: 'SQFT Price',
        accessor: 'sqftPrice',
        align: 'right'
      },
      {
        Header: '% of Valuation',
        accessor: 'valuationPercent',
        align: 'right'
      },
      {
        Header: 'Property Info',
        accessor: (row: PropertyInterface) => (
          <span
            className="link"
            onClick={() => {
              setHistoricalListingModal(true)
              setSelectedProperty(row)
            }}
          >
            Learn More
          </span>
        ),
        align: 'right'
      },
      {
        Header: 'Images',
        accessor: row => (
          <img
            className="link"
            src={`${process.env.baseUrl}/images/photos-link.svg`}
            onClick={() => {
              setGalleryModal(true)
              setSelectedProperty(row)
            }}
            alt="photos"
          />
        ),
        align: 'right'
      },
      {
        Header: '',
        accessor: 'order',
        align: 'right'
      }
    ])
    setData(tableData)
  }, [tableData])

  return (
    <div className="table-spacer">
      <Table columns={columns} data={data} fetchData={fetchData} width={1050} />
      {selectedProperty && historicalListingModal && (
        <HistoricalListingModal
          closeModal={() => {
            setHistoricalListingModal(false)
            setSelectedProperty(null)
          }}
          property={selectedProperty}
          type={type}
        />
      )}
      {selectedProperty && galleryModal && (
        <GalleryModal
          photos={selectedProperty.photos}
          closeModal={() => {
            setGalleryModal(false)
            setSelectedProperty(null)
          }}
        />
      )}
    </div>
  )
}
