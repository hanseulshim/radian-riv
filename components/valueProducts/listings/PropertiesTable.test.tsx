import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PropertiesTable from './PropertiesTable'

describe('PropertiesTable', () => {
  let component
  beforeAll(() => {
    const props = {
      tableData: [],
      type: 'Sold'
    }
    component = shallow(<PropertiesTable {...props} />)
  })
  it('should have container', () => {
    expect(component.find('.table-spacer')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
