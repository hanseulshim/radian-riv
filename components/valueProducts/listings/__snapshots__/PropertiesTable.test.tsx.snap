// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertiesTable should match snapshot 1`] = `
<div
  className="table-spacer"
>
  <Table
    columns={Array []}
    data={Array []}
    fetchData={[Function]}
    width={1050}
  />
</div>
`;
